File: src/main/g8/$service__packaged$/src/main/$java__packaged$/$servicePackage__packaged$/hello/HelloServiceAsyncImpl.java
package com.isuwang.soa.hello;

import com.isuwang.dapeng.core.SoaException;
import com.isuwang.soa.hello.domain.Hello;
import com.isuwang.soa.hello.service.HelloServiceAsync;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.Future;

public class HelloServiceAsyncImpl implements HelloServiceAsync {
    @Override
    public Future<String> sayHello(Hello hello, long timeout) throws SoaException {
        return CompletableFuture.supplyAsync(() -> hello.toString());
    }

    @Override
    public Future<Void> test(String content, long timeout) throws SoaException {
        return CompletableFuture.supplyAsync(() -> {
            System.out.println(" Hello world: " + content);
            return null;
        });
    }
}

File: src/main/g8/$service__packaged$/src/main/$scala__packaged$/$servicePackage__packaged$/scala/hello/HelloServiceAsyncImpl.scala
package com.isuwang.soa.scala.hello

import com.isuwang.soa.hello.scala.domain.Hello
import com.isuwang.soa.hello.scala.service.HelloServiceAsync
import scala.concurrent.ExecutionContext.Implicits.global

import scala.concurrent.Future

class HelloServiceAsyncImpl extends  HelloServiceAsync{
  /**
    *
    **/
  override def sayHello(hello: Hello, timeout: Long): Future[String] = {
    Future {s" hello: \${hello}, timeout: \${timeout} "}
  }

  /**
    *
    **/
  override def test(content: String, timeout: Long):Future[Unit] = {
    Future {println(s" content: \${content}, timeout: \${timeout}")}
  }
}